efunctions/efunctions_ecolors

```diff
@@ -20,13 +24,23 @@ else
 		BRACKET=`tput sgr0; tput bold; tput setaf 4`
 		NORMAL=`tput sgr0`
 	else
-		#echo "Fallback colours" 1>&2
-		GOOD="$(/bin/echo -e "\e[1;32m")"
-		WARN="$(/bin/echo -e "\e[1;33m")"
-		BAD="$(/bin/echo -e "\e[1;31m")"
-		HILITE="$(/bin/echo -e "\e[1;36m")"
-		BRACKET="$(/bin/echo -e "\e[1;34m")"
-		NORMAL="$(/bin/echo -e "\e[0m")"
+		if tty -s; then
```

Comment:
FYI `man tty` states `-s` is deprecated on OS X (BSD):

```
     -s      Do not write the terminal name; only the exit status is affected when this option is specified.  The -s option is
             deprecated in favor of the ``test -t 0'' command.
```

but no mention of that on Ubuntu 14.04.